### DISCUSSION TITLE
script failed: revert: Didn't send enough ether

### DISCUSSION THREAD

**Discussion by akashnikam25** [Cyfrin/foundry-full-course-cu#1652] [Q&A]
In First case I tried like this
```
address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment(
"Fundme",
block.chainid
);
Fundme fundme = new Fundme(payable(mostRecentlyDeployed));
http://fundme.fund/{value: 0.1 ether}();
```
error : Didn't send enough ether 

but in other case if I tried like this
```
Fundme(payable(mostRecentlyDeployed)).fund{value: 0.1 ether}();
```
In this case, it is working fine. In both cases, we are doing the same thing, but why is it not working in the first case?

**Comment by EngrPips** [comment]
Hello @akashnikam25, Please explain how this code `http://fundme.fund/{value: 0.1 ether}();` works to me.

**Reply by akashnikam25** [reply]
Fundme fundme = new Fundme(payable(mostRecentlyDeployed));
fundme.fund{value:1 ether}();
In this case, we have created a separate object and we are calling fund function. In that case i got this error script failed: revert: Didn't send enough ether.

but in the Second case, I didn't see any error.

can you paste your code?

**Reply by EngrPips** [reply]
Please provide the code that gives an error and provide the one that works fine. Put a tag on the codes to specify them so I can easily distinguish between them

**Reply by ighodaro-daniel** [reply]
pls help out , in section 2 the fundme contract , I  inputted more than 1ether in the value input when after i compiled my code but still get error : [vm]from: 0x5B3...eddC4to: FundMe.(constructor)value: 3000000000000000000 weidata: 0x608...90033logs: 0hash: 0x940...bf665
creation of FundMe errored: Error occurred: revert.

revert
	The transaction has been reverted to the initial state.
Note: The called function should be payable if you send value and the value you send should be less than your current balance.
You may want to cautiously increase the gas limit if the transaction went out of gas.

and here is my code: //SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

contract FundMe{
    function funds() payable public{
        require(msg.value > 1e18,"Insufficient Balance");
    }
}

**Reply by EngrPips** [reply]
```javascript
//SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

contract FundMe{
function funds() payable public{
require(msg.value > 1e18,"Insufficient Balance");
}
}
```

**Reply by EngrPips** [reply]
Hello @ighodaro-daniel, Providing more details would really make it easy for us to help you. share your deployment script and the full `FundMe` contract.
